### ISSUE TITLE
[BUG] MediaElement Android crashes when OnFullscreenButtonClick

### ISSUE THREAD

**Issue by vikher** [CommunityToolkit/Maui#2044]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Did you read the "Reporting a bug" section on Contributing file?

- [X] I have read the "Reporting a bug" section on Contributing file: https://github.com/CommunityToolkit/Maui/blob/main/CONTRIBUTING.md#reporting-a-bug

### Current Behavior

The Android app crashes when the fullscreen button is clicked.

### Expected Behavior

The app should not crash when the fullscreen button is clicked.

### Steps To Reproduce

Use MediaElement version 3.1.0.
Click the fullscreen button mainly on a Galaxy S23 Ultra, Galaxy S22 Ultra, or Pixel 8 Pro.

### Link to public reproduction project repository

https://github.com/vikher/testmedia

### Environment

```markdown
- .NET MAUI CommunityToolkit Version 9.0.2
- OS: net8.0-android
- .NET MAUI: MauiVersion 8.0.70
- CommunityToolkit.Maui.MediaElement Version 3.1.0
```

### Anything else?

- Occurs mostly on Galaxy S23 Ultra, Galaxy S22 Ultra, and Pixel 8 Pro.
- Reported by Visual Studio App Center.
- I'm using MediaElement version 3.1.0 for Android because:

1.   Version 3.1.1 breaks fullscreen functionality.
2.   Versions 4.0+ causes the app to reopen on the last visited page and continue background audio, behavior that is not desired.
3.  Awaiting approval of [CommunityToolkit/Maui PR #2039](https://github.com/CommunityToolkit/Maui/pull/2039) to use the latest NuGet package.

Error Details:

Error: System.ObjectDisposedException
Message: ObjectDisposed_Generic ObjectDisposed_ObjectName_Name, CommunityToolkit.Maui.Core.Views.MauiMediaElement

```
Xamarin Exception Stack:
System.ObjectDisposedException: ObjectDisposed_Generic
ObjectDisposed_ObjectName_Name, CommunityToolkit.Maui.Core.Views.MauiMediaElement
  at Java.Interop.JniPeerMembers.AssertSelf(IJavaPeerable )
  at Java.Interop.JniPeerMembers.JniInstanceMethods.InvokeVirtualVoidMethod(String , IJavaPeerable , JniArgumentValue* )
  at Android.Views.ViewGroup.AddView(View )
  at CommunityToolkit.Maui.Core.Views.MauiMediaElement.OnFullscreenButtonClick(Object sender, FullscreenButtonClickEventArgs e)
  at Com.Google.Android.Exoplayer2.UI.StyledPlayerView.IFullscreenButtonClickListenerImplementor.OnFullscreenButtonClick(Boolean isFullScreen)
  at Com.Google.Android.Exoplayer2.UI.StyledPlayerView.IFullscreenButtonClickListenerInvoker.n_OnFullscreenButtonClick_Z(IntPtr jnienv, IntPtr native__this, Boolean isFullScreen)
  at Android.Runtime.DynamicMethodNameCounter.24(IntPtr , IntPtr , Boolean )
```

Code Context:

I'm using DisconnectMediaElementHandler method to dispose the media element, based on [this discussion](https://github.com/CommunityToolkit/Maui/discussions/1663):

```
  public override void Stop()
  {
      base.Stop();
      DisconnectMediaElementHandler();
  }

private void DisconnectMediaElementHandler()
{
    mediaElement.Handler?.DisconnectHandler();
    mediaElement.Dispose();
}
```
The DisconnectMediaElementHandler method is executed when the page disappears, not when the fullscreen button is clicked, at least on Android.

I'm not  using ContentPage_Unloaded because, on iOS, when the fullscreen button is clicked and it enters fullscreen mode, CnkContentPage_Unloaded gets called and disposes of the media element view.

```
        //DO NOT DO THIS IN IOS
        void ContentPage_Unloaded(System.Object sender, System.EventArgs e)
        {
            DisconnectMediaElementHandler();
        }
```

Additional Information:
There is no current way to trigger a specialized event in Maui when the fullscreen button is clicked.

The following screenshot from the app center
![image](https://github.com/user-attachments/assets/83705a21-6a05-4580-8a4c-1420b2d52ce4)

**Comment by ne0rrmatrix** [comment]
There is a PR that came out last night that will add support for full screen event handling. https://github.com/CommunityToolkit/Maui/pull/2041

**Comment by ne0rrmatrix** [comment]
I tested against current main repo for MediaElement and toolkit. Could not reproduce issue. Will keep this open till #2041 is merged.

**Comment by ne0rrmatrix** [comment]
Can you try the latest version? 4.1.0 Came out last week.

**Comment by vikher** [comment]
After updating to MediaElement Version 4.1.0, I can confirm the error `System.ObjectDisposedException: ObjectDisposed_Generic ObjectDisposed_ObjectName_Name in CommunityToolkit.Maui.Core.Views.MauiMediaElement` 
```
Java.Interop.JniPeerMembers.AssertSelf(IJavaPeerable )
Java.Interop.JniPeerMembers.JniInstanceMethods.InvokeVirtualVoidMethod(String , IJavaPeerable , JniArgumentValue* )
Android.Views.ViewGroup.AddView(View )
CommunityToolkit.Maui.Core.Views.MauiMediaElement.OnFullscreenButtonClick(Object sender, FullscreenButtonClickEventArgs e)
Com.Google.Android.Exoplayer2.UI.StyledPlayerView.IFullscreenButtonClickListenerImplementor.OnFullscreenButtonClick(Boolean isFullScreen)
Com.Google.Android.Exoplayer2.UI.StyledPlayerView.IFullscreenButtonClickListenerInvoker.n_OnFullscreenButtonClick_Z(IntPtr jnienv, IntPtr native__this, Boolean isFullScreen)
```
has stopped occurring. However, I have encountered a new issue:

```
System.ArgumentNullException
System.ArgumentNullException: ArgumentNull_Generic Arg_ParamName_Name, intent
O
CommunityToolkit.Maui.Media.Services.MediaControlsService.OnStartCommand(Intent intent, StartCommandFlags flags, Int32 startId)
Android.App.Service.n_OnStartCommand_Landroid_content_Intent_II(IntPtr jnienv, IntPtr native__this, IntPtr native_intent, Int32 native_flags, Int32 startId)
Android.Runtime.JNINativeWrapper.Wrap_JniMarshal_PPLII_I(_JniMarshal_PPLII_I callback, IntPtr jnienv, IntPtr klazz, IntPtr p0, Int32 p1, Int32 p2)
```


I noticed that your team closed issue https://github.com/CommunityToolkit/Maui/issues/2163 with a similar stack trace. Therefore, I will reopen the bug and try to provide steps to reproduce the issue.
I’ve added a screenshot from App Center as evidence of this new bug. Let me know if you need any further details.

![image](https://github.com/user-attachments/assets/25963411-558c-402e-b551-1ab0039be3bd)

**Comment by ne0rrmatrix** [comment]
Can u test against this PR: https://github.com/CommunityToolkit/Maui/pull/2076 and check and see if it fixes your issue. Based on what you have posted without testing it is the service crashing. That issue has been addressed in current PR listed above. That is waiting on review. If I can have you check that bug and verify it is fixed that would be helpful.

**Comment by brminnick** [comment]
> System.ArgumentNullException
System.ArgumentNullException: ArgumentNull_Generic Arg_ParamName_Name, intent
O
CommunityToolkit.Maui.Media.Services.MediaControlsService.OnStartCommand(Intent intent, StartCommandFlags flags, Int32 startId)
Android.App.Service.n_OnStartCommand_Landroid_content_Intent_II(IntPtr jnienv, IntPtr native__this, IntPtr native_intent, Int32 native_flags, Int32 startId)
Android.Runtime.JNINativeWrapper.Wrap_JniMarshal_PPLII_I(_JniMarshal_PPLII_I callback, IntPtr jnienv, IntPtr klazz, IntPtr p0, Int32 p1, Int32 p2)

@vikher Would you please open a new [BUg Report Issue](https://github.com/CommunityToolkit/Maui/issues/new?assignees=&labels=bug%2Cunverified&projects=&template=bug_report_template.yml&title=%5BBUG%5D+) for this? We'll need you to provide us a reproduction sample to help us narrow down the problem and solve it.

The `ArgumentNullException` is thrown [here](https://github.com/CommunityToolkit/Maui/blob/e6a170e22e1513fb86fa46d1a11fae533a7b7d85/src/CommunityToolkit.Maui.MediaElement/Services/MediaControlsService.android.cs#L46) when Android passes in an `Intent` that is `null`. I'm surprised that a `null` Intent is being passed in to `OnStartCommand()` when the app is running in the foreground; I assumed this would only happen on a BackgroundService.

Here's what the Android docs say about `OnStartCommand()`:

 https://developer.android.com/develop/background-work/services

> When a service is started, it has a lifecycle that's independent of the component that started it. The service can run in the background indefinitely, even if the component that started it is destroyed. As such, the service should stop itself when its job is complete by calling [stopSelf()](https://developer.android.com/reference/android/app/Service#stopSelf()), or another component can stop it by calling [stopService()](https://developer.android.com/reference/android/content/Context#stopService(android.content.Intent)).
> 
> An application component such as an activity can start the service by calling [startService()](https://developer.android.com/reference/android/content/Context#startService(android.content.Intent)) and passing an [Intent](https://developer.android.com/reference/android/content/Intent) that specifies the service and includes any data for the service to use. The service receives this [Intent](https://developer.android.com/reference/android/content/Intent) in the [onStartCommand()](https://developer.android.com/reference/android/app/Service#onStartCommand(android.content.Intent,%20int,%20int)) method.

**Comment by brminnick** [comment]
> That is the only place that service can be started now. It does get called in more than one place. But it will always check to see if it has authorization first now. So no null intent. Only way that could happen is if notification permission were revoked while it was playing video. I believe that would close the notification too.

The key part we've overlooked is that the Android Operating System can start the service. In other words, it's not just our code invoking this method.

> An application component such as an activity can start the service by calling [startService()](https://developer.android.com/reference/android/content/Context#startService(android.content.Intent)) and passing an [Intent](https://developer.android.com/reference/android/content/Intent) that specifies the service and includes any data for the service to use. The service receives this [Intent](https://developer.android.com/reference/android/content/Intent) in the [onStartCommand()](https://developer.android.com/reference/android/app/Service#onStartCommand(android.content.Intent,%20int,%20int)) method.

Let's pause the discussion since it is unrelated to this Issue and continue the discussion once @vikher has opened a new [Bug Report Issue](https://github.com/CommunityToolkit/Maui/issues/new?assignees=&labels=bug%2Cunverified&projects=&template=bug_report_template.yml&title=%5BBUG%5D+).

**Comment by ne0rrmatrix** [comment]
This bug report is stale. It points to a no longer supported version of dotnet and maui. The underlying issue has been addressed with updating to latest which has reworked how the service works. The libraries used in the affected version are deprecated and have been removed from media element. Try version 6.0 of media element and it has the same issue please open a new bug report.
